[Svelte] Add support for Svelte 5

Kocal opened the pull request:
| Q              | A
| -------------- | ---
| Bug fix?       | no
| New feature?   | yes <!-- please update src/**/CHANGELOG.md files -->
| Deprecations?  | no <!-- if yes, also update UPGRADE-*.md and src/**/CHANGELOG.md -->
| Documentation? | no <!-- required for new features, or documentation updates -->
| Issues         | Fix #2540 <!-- prefix each issue number with "Fix #", no need to create an issue if none exist, explain below instead -->
| License        | MIT

<!--
Replace this notice by a description of your feature/bugfix.
This will help reviewers and should be a good start for the documentation.

Additionally (see https://symfony.com/releases):
 - Always add tests and ensure they pass.
 - For new features, provide some code snippets to help understand usage.
 - Features and deprecations must be submitted against branch main.
 - Update/add documentation as required (we can help!)
 - Changelog entry should follow https://symfony.com/doc/current/contributing/code/conventions.html#writing-a-changelog-entry
 - Never break backward compatibility (see https://symfony.com/bc).
-->

Related to https://github.com/symfony/ux/pull/2288


This is in work-in-progress, I still have an error in the browser while trying to render the `Hello` component when using the AssetMapper: 
```
stimulus.index-S4zNcea.js:7 Error connecting controller

TypeError: Cannot read properties of undefined (reading 'call')
    at _n (client-BATiWue.js:7:27775)
    at client-BATiWue.js:7:39919
    at Hello (Hello-s876KnU.js:8:12)
    at svelte.index-TK8_2BG.js:7:28981
    at svelte.index-TK8_2BG.js:7:11011
    at Ht (svelte.index-TK8_2BG.js:7:24452)
    at Kt (svelte.index-TK8_2BG.js:7:25511)
    at dt (svelte.index-TK8_2BG.js:7:21898)
    at pt (svelte.index-TK8_2BG.js:7:22380)
    at svelte.index-TK8_2BG.js:7:11003
```

I didn't check how it deals with Webpack Encore yet (opened https://github.com/symfony/webpack-encore/issues/1384).

Kocal committed:
[Svelte] Add support for Svelte 5

maxperei commented:
hi there, this is one of a kind.. i had a look recently about svelte 5/asset mapper and i ran into the same, so following what i thought i wrote a `rollup.config.js` (inspired from [this repo](https://github.com/crisward/svelte-shared/commit/78bc5fb8bbc46fb950f3c0868f609777cdca2bc5)) on top of this branch to put all the things together
after a while and some distorsions ^^ i managed to render something but it's a big mess at least
anyway i feel this is not the right way things should be done with assetmapper as i got to use a module bundler
there is already a step to compile `.svelte` files and it's quite unpleasant tho
i could give extra details if it sounds interesting

Kocal commented:
Hey @maxperei, sure, any help would be appreciated! :)

maxperei commented:
ok well, i generated a [patch](https://github.com/user-attachments/files/23653721/fix__bundlize_svelte5_locally_for_assetmapper.patch) to give a try
of course you'll have to `npm i && npm run build` again (as well as `php bin/console assetmap:install` if necessary)
then if you go into `./assets/vendor/svelte` you see a new file : `index-client.js` and a bunch of other files like `svelte.index.js`, `internal/client.js` which are overridden and the left ones remain unchanged
this particular `index-client.js` is very tricky because *sometimes* it throws a 404 and you will have to expose this file into `public/` folder using `php bin/console asset-map:compile` or just copy it, and *sometimes* (maybe it's symfony cache or something i can't explain --') it works without exposing it

Kocal commented:
Hey @maxperei and sorry for the late reply, support for Svelte 5 is something I've put on the side for a while, but I'd really like to finish this task.

I tried your solution, which is indeed super hacky, but it's still one solution among... 0 other solutions 😅 

I've just rebased and force-pushed this PR, and moved the ux.symfony.com things to https://github.com/symfony/ux.symfony.com/pull/32, the files `svelte/internal/client.js` and `svelte/svelte.index.js` looks better, but it contains an unresolved dependency to `custom-element.js` (even if the file exists... maybe the AssetMapper `import` resolver is missing it?)

<img width="710" height="45" alt="image" src="https://github.com/user-attachments/assets/8b36004e-1b40-4e82-9946-2202a6d1519f" />

I will try to open an issue on Svelte to see if they can provide a JSDeliver/ESM-compatible build, that will be easier for everyone.

ChqThomas commented:
Hey @Kocal

I came across this post on X which reminded me of the issue we had when trying to integrate Svelte 5 in the initial PR:
https://x.com/joyofcodedev/status/2038259907451928807

I tried applying the same importmap approach with esm.sh cdn, and it seems to work in a minimal setup with a pre-built component:
https://codesandbox.io/p/devbox/svelte-importmap-forked-q8858g?workspaceId=ws_CJ7KxeYW5eGLozWC7m7Lw5

This might be a good starting point for a solution!

Kocal commented:
Hi, and thanks for the suggestions! 

Unfortunately, after several months of failed attempts, I think we’re simply going to drop support for Svelte 5 with AssetMapper in UX Svelte 3. 

Their CDN build is completely broken, and I’d rather have nothing (or at least, just support with Webpack Encore) than have to go to great lengths to try and get it working with AssetMapper. 

If anyone in the future decides to add Svelte 5 support for AssetMapper by offering a clean solution, it will be a pleasure!

Kocal commented:
I will open a new PR.
